src/test/java/com/tuplejump/stargate/util/Record.java

```diff
@@ -0,0 +1,101 @@
+package com.tuplejump.stargate.util;
+
+import com.datastax.driver.core.Row;
+
+import java.lang.reflect.Array;
+import java.util.*;
+
+public class Record {
+
+    private Map recordDefinition = new HashMap<String, String>();
+    private Map<String, Object> record = new HashMap<String, Object>();
+
+    public Record(String[] fields, Object[] values, String[] type) {
+        if (fields.length == values.length) {
+            for (int i = 0; i < fields.length; i++) {
+                recordDefinition.put(fields[i].toLowerCase(), type[i]);
+                record.put(fields[i].toLowerCase(), values[i]);
+            }
+        }
+    }
+
+    public Record(Row row, String indexCol) {
+        row.getColumnDefinitions().iterator().forEachRemaining(field -> {
+            String col = field.getName();
+            record.put(col.toLowerCase(), row.getObject(col));
+        });
+        record.remove(indexCol);
+    }
+
+    public String getInsertString() {
+        Iterator it = record.entrySet().iterator();
+        List<String> fieldList = new ArrayList<String>();
+        List<Object> valueList = new ArrayList<Object>();
+        List<String> types = new ArrayList<String>();
+        while (it.hasNext()) {
+            Map.Entry map = (Map.Entry) it.next();
+            fieldList.add((String) map.getKey());
+            valueList.add(map.getValue());
+            types.add((String) recordDefinition.get(map.getKey()));
+        }
+        return "(" + mkString(fieldList) + ")values(" + mkString(valueList, types) + ");";
+    }
+
+    private String mkString(List<String> list) {
```

Comment:
method name is not conveying what we are doing. Having the same one for two methods which are not really related, is also further misleading.